release/models/platform/openconfig-platform-types.yang

```diff
@@ -360,4 +383,56 @@ module openconfig-platform-types {
       "A generic type reflecting whether a hardware component
       is powered on or off";
   }
+
+  identity COMPONENT_SWITCHOVER_REASON {
+    description
+    "For components that have redundant roles (e.g. two
+    supervisors in a device, one as master the other as backup),
+    this is the base entity for the role switchover reason of
+    the component.";
+  }
+
+  identity SWITCHOVER_USER_INITIATED {
+    base COMPONENT_SWITCHOVER_REASON;
+    description
+      "User initiated the switchover of the componenent's
+      master/backup role.";
+  }
+
+  identity SWITCHOVER_LOST_MASTER {
+    base COMPONENT_SWITCHOVER_REASON;
+    description
+      "The component assumes master role due
+      to lost of communication with component of the master role.";
+  }
+
```

Comment:
Thanks. It is a good point that different system can have different failure reasons due to different implementations. 

I have restructured the proposal to use a container for help record different switchover scenarios. Can you help take a look? Thanks. 

Resolving this conversation because the anchored code has been deleted.